models/webhook.go

```diff
@@ -753,7 +756,21 @@ func (t *HookTask) deliver() {
 	t.IsDelivered = true
 
 	timeout := time.Duration(setting.Webhook.DeliverTimeout) * time.Second
-	req := httplib.Post(t.URL).SetTimeout(timeout, timeout).
+
+	var req *httplib.Request
+	if t.HTTPMethod == "post" {
+		req = httplib.Post(t.URL)
+		switch t.ContentType {
+		case ContentTypeJSON:
+			req = req.Header("Content-Type", "application/json").Body(t.PayloadContent)
+		case ContentTypeForm:
+			req.Param("payload", t.PayloadContent)
+		}
+	} else {
+		req = httplib.Get(t.URL).Param("payload", t.PayloadContent)
```

Comment:
Wouldn't payload be too big in such case?

Reply:
As @shilch mentioned: the get parameters are not limited by the spec. Browsers are no matter of concern since these request will be server2server.